src/lib/utils/utils.js

```diff
@@ -390,6 +390,12 @@ function sha3(arg) {
   return Web3.utils.sha3(arg);
 }
 
+function sha512(arg) {
+  const crypto = require('crypto');
+  const hash = crypto.createHash('sha512');
+  return hash.update(arg, 'utf8').digest('hex');
```

Comment:
Here, just as a precaution, you want to make sure that `arg` is a string, so we might want it to read

```js
return hash.update(arg.toString(), 'utf8').digest('hex');
```

Reply:
https://nodejs.org/dist/latest-v10.x/docs/api/crypto.html#crypto_hash_update_data_inputencoding

Maybe we should just throw if it's not a string?

```
> ({a: 1}).toString()
'[object Object]'
> ({a: 2}).toString()
'[object Object]'
```